activesupport/test/broadcast_logger_test.rb

```diff
@@ -3,75 +3,147 @@
-      assert_equal 10, log1.formatter
-      assert_equal 10, log2.formatter
+      logger.formatter = ::Logger::FATAL
+
+      assert_equal ::Logger::FATAL, log1.formatter
+      assert_equal ::Logger::FATAL, log2.formatter
+    end
+
+    test "#local_level= assigns the local_level to all loggers" do
+      assert_equal ::Logger::DEBUG, logger.local_level
+      logger.local_level = ::Logger::FATAL
+
+      assert_equal ::Logger::FATAL, log1.local_level
+      assert_equal ::Logger::FATAL, log2.local_level
+    end
+
+    test "#silence silences all loggers below the default level of ERROR" do
+      logger.silence do
+        logger.debug "test"
+      end
+
+      assert_equal [], log1.adds
+      assert_equal [], log2.adds
+    end
+
+    test "#silence does not silence at or above ERROR" do
+      logger.silence do
+        logger.error "from error"
+        logger.unknown "from unknown"
+      end
+
+      assert_equal [[::Logger::ERROR, "from error", nil], [::Logger::UNKNOWN, "from unknown", nil]], log1.adds
+      assert_equal [[::Logger::ERROR, "from error", nil], [::Logger::UNKNOWN, "from unknown", nil]], log2.adds
+    end
+
+    test "#silence allows you to override the silence level" do
+      logger.silence(::Logger::FATAL) do
+        logger.error "unseen"
+        logger.fatal "seen"
+      end
+
+      assert_equal [[::Logger::FATAL, "seen", nil]], log1.adds
+      assert_equal [[::Logger::FATAL, "seen", nil]], log2.adds
     end
 
     class FakeLogger
+      include LoggerSilence
+
       attr_reader :adds, :closed, :chevrons
-      attr_accessor :level, :progname, :formatter
+      attr_accessor :level, :progname, :formatter, :local_level
 
       def initialize
-        @adds      = []
-        @closed    = false
-        @chevrons  = []
-        @level     = nil
-        @progname  = nil
-        @formatter = nil
+        @adds        = []
+        @closed      = false
+        @chevrons    = []
+        @level       = ::Logger::DEBUG
+        @local_level = ::Logger::DEBUG
+        @progname    = nil
+        @formatter   = nil
+      end
+
+      def debug(message, &block)
+        add(::Logger::DEBUG, message, &block)
+      end
+
+      def info(message, &block)
+        add(::Logger::INFO, message, &block)
+      end
+
+      def warn(message, &block)
+        add(::Logger::WARN, message, &block)
+      end
+
+      def error(message, &block)
+        add(::Logger::ERROR, message, &block)
+      end
+
+      def fatal(message, &block)
+        add(::Logger::FATAL, message, &block)
       end
 
-      def debug msg, &block
-        add(:omg, nil, msg, &block)
+      def unknown(message, &block)
+        add(::Logger::UNKNOWN, message, &block)
       end
 
       def << x
         @chevrons << x
       end
 
-      def add(*args)
-        @adds << args
+      def add(message_level, message=nil, progname=nil, &block)
+        @adds << [message_level, message, progname] if message_level >= local_level
```

Comment:
Don't love this, because we're basically testing the behaviour of a logger of by reimplementing some of the logic.